gtsam/gtsam.i

```diff
@@ -2759,21 +2759,29 @@ virtual class EssentialMatrixFactor : gtsam::NoiseModelFactor {
 };
 
 #include <gtsam/slam/dataset.h>
+
 class SfmTrack {
+  SfmTrack();
   Point3 point3() const;
   size_t number_measurements() const;
+  void set_point3(gtsam::Point3& p_);
   pair<size_t, gtsam::Point2> measurement(size_t idx) const;
   pair<size_t, size_t> siftIndex(size_t idx) const;
+  void add_measurement(const pair<size_t, gtsam::Point2>& m);
 };
 
 class SfmData {
+  SfmData();
   size_t number_cameras() const;
   size_t number_tracks() const;
   gtsam::PinholeCamera<gtsam::Cal3Bundler> camera(size_t idx) const;
   gtsam::SfmTrack track(size_t idx) const;
+  void add_track(const gtsam::SfmTrack& t) ;
+  void add_camera(const gtsam::SfmCamera& cam);
 };
 
 gtsam::SfmData readBal(string filename);
+bool writeBAL(string filename, gtsam::SfmData& data);
```

Comment:
It looks like the filename argument is a `const std::string& ` in dataset.h

Reply:
that is repeated across a lot of functions(eg readBal, load2D, etc). I'm not sure why it is being used the way it is. Maybe @ProfFan could shed some light on it?

Reply:
I think this is just history :)

Reply:
const and & are mostly ignored in .i files. Just using string is fine.